packages/popover/src/Popover.styles.ts

```diff
@@ -78,33 +95,91 @@ const transformOriginStyles: Record<ExtendedPlacement, string> = {
   `,
 };
 
-const getTransformStyles = (placement: ExtendedPlacement, spacing: number) =>
-  cx({
-    [css`
+const getClosedStyles = (placement: ExtendedPlacement, spacing: number) => {
+  if (placement.startsWith('top')) {
+    return css`
+      opacity: 0;
       transform: translateY(${spacing}px) scale(0);
-    `]: placement.startsWith('top'),
-    [css`
+    `;
+  }
+
+  if (placement.startsWith('bottom')) {
+    return css`
+      opacity: 0;
       transform: translateY(-${spacing}px) scale(0);
-    `]: placement.startsWith('bottom'),
-    [css`
+    `;
+  }
+
+  if (placement.startsWith('left')) {
+    return css`
+      opacity: 0;
       transform: translateX(${spacing}px) scale(0);
-    `]: placement.startsWith('left'),
-    [css`
+    `;
+  }
+
+  if (placement.startsWith('right')) {
+    return css`
+      opacity: 0;
       transform: translateX(-${spacing}px) scale(0);
-    `]: placement.startsWith('right'),
-    [css`
-      transform: scale(0);
-    `]: placement.startsWith('center'),
-  });
+    `;
+  }
 
-const closedStyles = css`
-  opacity: 0;
-`;
+  return css`
+    opacity: 0;
+    transform: scale(0);
+  `;
+};
 
-const openStyles = css`
-  opacity: 1;
-  pointer-events: initial;
-`;
+const getOpenStyles = (placement: ExtendedPlacement) => {
+  if (placement.startsWith('top')) {
+    return css`
+      opacity: 1;
+      pointer-events: initial;
+      transform: translateY(0) scale(1);
+    `;
+  }
+
+  if (placement.startsWith('bottom')) {
+    return css`
+      opacity: 1;
+      pointer-events: initial;
+      transform: translateY(0) scale(1);
+    `;
+  }
+
+  if (placement.startsWith('left')) {
+    return css`
+      opacity: 1;
+      pointer-events: initial;
+      transform: translateX(0) scale(1);
+    `;
+  }
+
+  if (placement.startsWith('right')) {
+    return css`
+      opacity: 1;
+      pointer-events: initial;
+      transform: translateX(0) scale(1);
+    `;
+  }
+
+  return css`
+    opacity: 1;
+    pointer-events: initial;
+    transform: scale(1);
+  `;
+};
+
+const getTransitionStyles = (
+  placement: ExtendedPlacement,
+  spacing: number,
+) => ({
+  exited: getClosedStyles(placement, spacing),
+  entering: getClosedStyles(placement, spacing),
+  entered: getOpenStyles(placement),
+  exiting: getClosedStyles(placement, spacing),
+  unmounted: getClosedStyles(placement, spacing),
```

Comment:
I noticed a slight delay after clicking before the popover started transitioning. I wonder if `entering` should have `getOpenStyles`?